internal/pkg/install/mount.go

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package install
+
+import (
+	"log"
+
+	"github.com/autonomy/talos/internal/pkg/blockdevice"
+	"github.com/autonomy/talos/internal/pkg/blockdevice/probe"
+	gptpartition "github.com/autonomy/talos/internal/pkg/blockdevice/table/gpt/partition"
+	"github.com/autonomy/talos/internal/pkg/blockdevice/util"
+	"github.com/autonomy/talos/internal/pkg/constants"
+	"github.com/autonomy/talos/internal/pkg/mount"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+func Mount() (err error) {
+	log.Println("Discovering mountpoints")
+	var mp *mount.Points
+	if mp, err = mountpoints(); err != nil {
+		return errors.Errorf("error initializing block devices: %v", err)
+	}
+
+	/*
+		// This seems to be corrupting the filesystem
+		// Running xfs_repair finds a bunch of badness
+
+		log.Println("Repairing mountpoints")
+		for _, name := range []string{constants.RootPartitionLabel, constants.DataPartitionLabel, constants.BootPartitionLabel} {
+
+			if mountpoint, ok := mp.Get(name); ok {
+				if err = repair(mountpoint); err != nil {
+					return errors.Errorf("error fixing %s partition: %v", name, err)
+				}
+			}
+		}
+	*/
+
+	log.Println("Attempting to mount filesystems")
+	iter := mp.Iter()
+	for iter.Next() {
+		if err = mount.WithRetry(iter.Value(), mount.WithPrefix(constants.NewRoot)); err != nil {
+			return errors.Errorf("error mounting partitions: %v", err)
+		}
+	}
+	if iter.Err() != nil {
+		return iter.Err()
+	}
+
+	/*
+		log.Println("Attempting to grow data partition")
+		if mountpoint, ok := mp.Get(constants.DataPartitionLabel); ok {
+			// NB: The XFS partition MUST be mounted, or this will fail.
+			if err = xfs.GrowFS(path.Join(constants.NewRoot, mountpoint.Target())); err != nil {
+				return errors.Errorf("error growing data partition file system: %v", err)
+			}
+		}
+	*/
```

Comment:
This will be a blocker to merging this until we can figure out why the corruption. Reason being is that the Talos images are built as small as possible, with little to no room for anything. So the node will almost immediately fail with a full disk. I can work on finding why the corruption happens 👍 .